src/umpire/Umpire.hpp

```diff
@@ -87,7 +87,7 @@ void print_allocator_records(Allocator allocator, std::ostream& os = std::cout);
  *
  * \param allocator source Allocator.
  */
-std::vector<const util::AllocationRecord*> get_allocator_records(Allocator allocator);
+std::vector<util::AllocationRecord> get_allocator_records(Allocator allocator);
```

Comment:
I don’t mind using pointers, but the problem came about with the `ResourceManager::deregisterAllocation()`, which had originally returned a pointer. If the `AllocationMap` internally manages its own data (a switch this PR makes), then the pointer isn't valid after the de-registration. `ResourceManager::findAllocationRecord()`, and `AllocationMap::find*()` still return pointers.

If you'd rather this method return a vector of pointers, I think that can work since the allocations are still registered by the `AllocationMap` and that class doesn't move around memory internally

Reply:
Okay, this makes sense.  I can't remember what the justification was for using pointers in the first place (@davidbeckingsale?).  Since this PR improves performance by so much, am fine with these changes and we can pick up the pointer thread later if/when we want to deal with improvements to the right of the decimal point.